src/parcsr_ls/par_ilu_setup.c

```diff
@@ -271,12 +271,12 @@ hypre_ILUSetup( void               *ilu_vdata,
    }
    if(matD)
    {
-      hypre_TFree(matD, HYPRE_MEMORY_HOST);
+      hypre_TFree(matD, HYPRE_MEMORY_DEVICE);
```

Comment:
@oseikuffuor1 This fix is correct. But how could it work before?

Reply:
@Hitenze is this matD constructed on the host only? Do we need it to be `HYPRE_MEMORY_DEVICE` or just `HOST`? I am not sure. Thanks!

Reply:
@liruipeng I think this was not failing earlier since it was only used outside HYPRE_USING_CUDA. So DEVICE Was still the CPU and everything was fine. Waiting to hear back from Tianshi as to whether matD and matMD should really be on DEVICE or HOST

Reply:
> @liruipeng I think this was not failing earlier since it was only used outside HYPRE_USING_CUDA. So DEVICE Was still the CPU and everything was fine. Waiting to hear back from Tianshi as to whether matD and matMD should really be on DEVICE or HOST

But you ran GPU tests. So it was not part of it?

Reply:
@liruipeng @oseikuffuor1 Yes we should make this change. I think the reason previous version works is that we are not actually reaching this part of the code. We'll reach this part only if we pass a already allocated ilu_data to the ilu setup function. In the hypre_ILUDestroy function we free it with HYPRE_MEMORY_DEVICE. What do you think?

Reply:
Thanks @Hitenze. It looks like everything checks out with the current change. matD is not used when a DEVICE ILU setup is called, so we don't have an issue on the GPU.